components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/DefaultGroupIDExtractorImpl.java

```diff
@@ -5,14 +5,15 @@
 import org.json.JSONException;
 import org.json.JSONObject;
 import org.wso2.carbon.apimgt.api.LoginPostExecutor;
+import org.wso2.carbon.apimgt.api.NewPostLoginExecutor;
 import org.wso2.carbon.apimgt.impl.internal.ServiceReferenceHolder;
 import org.wso2.carbon.base.MultitenantConstants;
 import org.wso2.carbon.user.core.UserRealm;
 import org.wso2.carbon.user.core.UserStoreManager;
 import org.wso2.carbon.user.core.service.RealmService;
 import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
 
-public class DefaultGroupIDExtractorImpl implements LoginPostExecutor {
+public class DefaultGroupIDExtractorImpl implements NewPostLoginExecutor {
```

Comment:
is "NewPostLoginExecutor" a better naming for the interface? Because what will happen if another new one come in future? 
my two cents..

Reply:
I added that name because when someone is going to write their own implementation to fetch groupid's, by looking at the name they will get an idea on which interface to use. If there is a better suggestion will change :) .